src/pkg_exporter/textfile.py

```diff
@@ -50,12 +59,91 @@ def main():
     rebootmanager.query()
     reboot_gauge.set(rebootmanager.getMetricValue())
 
-    exporter_file = os.getenv("PKG_EXPORTER_FILE", "/var/prometheus/pkg-exporter.prom")
+
+def write_registry_to_file(registry, exporter_file=None):
+    if not exporter_file:
+        exporter_file = os.getenv(
+            "PKG_EXPORTER_FILE", "/var/prometheus/pkg-exporter.prom"
+        )
     exporter_dir = os.path.dirname(exporter_file)
     os.makedirs(exporter_dir, exist_ok=True)
 
     write_to_textfile(exporter_file, registry)
 
 
+def serve(addr, port, timewait, rootdir):
+    start_http_server(addr=addr, port=port)
+    while True:
+        sleep(timewait)
+        populate_registry(rootdir)
+
+
+def processArgs():
+    parser = argparse.ArgumentParser(
+        description="Collect metrics from apt and export it as a service"
+    )
+    group = parser.add_mutually_exclusive_group()
+
+    group.add_argument(
+        "-f",
+        "--exporter-file",
+        type=str,
+        default=os.getenv(
+            "PKG_EXPORTER_FILE",
+            "/var/prometheus/pkg-exporter.prom"),
+        help="File to export, if used the content will not be served",
```

Comment:
In my opinion, the default is "output to a file" anyway? The comment indicates that by default the HTTP Server is run.